aws/resource_aws_db_parameter_group_test.go

```diff
@@ -132,8 +148,19 @@ func TestAccAWSDBParameterGroup_basic(t *testing.T) {
 					testAccCheckAWSDBParameterGroupAttributes(&v, groupName),
 					testAccCheckAWSDBParameterNotUserDefined(resourceName, "collation_connection"),
 					testAccCheckAWSDBParameterNotUserDefined(resourceName, "collation_server"),
-					resource.TestCheckNoResourceAttr(resourceName, "parameter.2475805061.value"),
-					resource.TestCheckNoResourceAttr(resourceName, "parameter.1706463059.value"),
+					resource.TestCheckResourceAttr(resourceName, "parameter.#", "3"),
+					test.TestCheckTypeSetElemNestedAttrs(resourceName, "parameter.*", map[string]string{
+						"name":  "character_set_results",
+						"value": "utf8",
+					}),
+					test.TestCheckTypeSetElemNestedAttrs(resourceName, "parameter.*", map[string]string{
+						"name":  "character_set_server",
+						"value": "utf8",
+					}),
+					test.TestCheckTypeSetElemNestedAttrs(resourceName, "parameter.*", map[string]string{
+						"name":  "character_set_client",
+						"value": "utf8",
+					}),
```

Comment:
This seems to be changing the test, since there were two items, and no values set for `value`, and now there are three and they have values.. Is that intentional?

Reply:
Ah yes this is a bit quirky. There isn't really anyway to represent a specific set element that doesn't exist... My PR proposes helpers to guarantee a single element meets the desired criteria. So I flipped the test to "3 elements with the following property maps, and that the set only contains 3 entries (and not 5)"

Reply:
I inferred that this is what was being validated by looking at the test config (goes from 5 parameters to 3)